lib/decorators/party_name_attributes.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+require 'scraped'
+
+# Members on the members page are grouped by party but the names or ids of the
+# parties are not given. Instead, each party is signified by an image.
+
+# This decorator maps the party name and id to the image name and populates the
+# party image <img> tags with party_name and party_id attributes.
+class PartyNameAttributes < Scraped::Response::Decorator
+  PARTY_IMAGES_MAPPED = {
+    'pri01.png'                     => ['Partido Revolucionario Institucional', 'PRI'],
+    'pan.png'                       => ['Partido Acción Nacional', 'PAN'],
+    'prd01.png'                     => ['Partido de la Revolución Democrática', 'PRD'],
+    'logvrd.jpg'                    => ['Partido Verde Ecologista', 'PVE'],
+    'logo_movimiento_ciudadano.png' => ['Movimiento Ciudadano', 'MC'],
+    'logpt.jpg'                     => ['Partido del Trabajo', 'PT'],
+    'panal.gif'                     => ['Partido Nueva Alianza', 'PANAL'],
+    'LogoMorena.jpg'                => ['Movimiento Regeneración Nacional', 'MORENA'],
+    'encuentro.png'                 => ['Encuentro Social', 'PES'],
+    'logo_SP.jpg'                   => ['Sin Partido', '_IND'],
+    'independiente.png'             => ['Sin Partido', '_IND'],
+  }.freeze
+
+  def body
+    Nokogiri::HTML(super).tap do |doc|
+      doc.css('img').each do |img|
+        key = img.attr('src').split('/').last
+        next unless PARTY_IMAGES_MAPPED.key?(key)
+        img[:party_id] = PARTY_IMAGES_MAPPED[key].last
+        img[:party_name] = PARTY_IMAGES_MAPPED[key].first
```

Comment:
Checking if there _is_ a key and then looking up by it is a little bit repetitive. I'd suggest something like this would be a little bit clearer and simpler:

```ruby
        party_data = PARTY_IMAGES_MAPPED[key] or next
        img[:party_name] = party_data.first
        img[:party_id] = party_data.last
```

(though `PARTY_IMAGES_MAPPED` is a slightly odd name. `PARTY_IMAGE_MAP` would be clearer too IMO)